Add hook and mocked data for testing API requests.

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Add a hook and mocked data for testing API requests.
Based on @mikkamp's snippet. 

Maybe we can add this hook permanently, and keep mocks in `/tests` folder, to make testing PRs and features easier.

Maybe more long-term alternative to mocks like this https://github.com/woocommerce/google-listings-and-ads/compare/feature/244-connect-data...feature/244-products-report-mock-local


### Screenshots:

<!--- Optional --->


### Detailed test instructions:

1. Checkout this branch
2. Add a code snippet:
	  ```php
	  add_filter(
		  'gla_prepared_response_reports',
		  function( $data, $classname ) {
			  if ( preg_match( '/MerchantCenter.ReportsController/', $classname ) ) {
				  return json_decode( file_get_contents( __DIR__ . '/tests/mocks/mc/report.json' ), true ) ?: [];
			  }
			  if ( preg_match( '/Ads.ReportsController/', $classname ) ) {
				  return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/report.json' ), true ) ?: [];
			  }
			  return $data;
		  },
		  10,
		  2
	  );
	  ```
3. Test the PR https://github.com/woocommerce/google-listings-and-ads/pull/569


### Changelog Note:

> Add `gla_prepared_response_{schema}` filter.

tomalec committed:
Add hook and mocked data for testing API requests.

eason9487 reviewed (approved):
Tested locally and works well. Thank you for this! 😄

mikkamp committed:
Use route name in prepared response filter

tomalec committed:
Move mock repsponses to directories that matches closer the endpoint path.

tomalec committed:
Add a short description and how-to for mocking.

Mention caveats from https://github.com/woocommerce/google-listings-and-ads/pull/575#issuecomment-838324289
